### PR TITLE
Small improvement in the granularity example

### PR DESCRIPTION
**Pull request by jcranky**

### CODE REVIEW

**Comment by jcranky** [comment]
I'm not sure if this is ideal, but the goal was to bring the "bad" and the "good" examples closer in meaning.

File: site/src/main/tut/datatypes/io.md [hunk 1]
```diff
@@ -1152,19 +1152,21 @@ IO {
 }
 ```
 
-In FP we embrace reasoning about our programs and since `IO` is a `Monad` you can compose bigger programs from small ones in a `for-comprehention` for example:
+In FP we embrace reasoning about our programs and since `IO` is a `Monad` you can compose bigger programs from small ones in a `for-comprehention`.
+For example:
```
Comment (ChristopherDavenport) [inline site/src/main/tut/datatypes/io.md]: We still have 2 standards due to the example above using semicolon in line. Likely we should pick a direction we like and be consistent.

File: site/src/main/tut/datatypes/io.md [hunk 2]
```diff
@@ -1152,19 +1152,21 @@ IO {
...
+    _    <- writeToDatabase(data)
+    _    <- sendBytesOverTcp(data)
+    _    <- launchMissiles
   } yield ()
 ```
 
-Here you have a simple prompt program that is, at the same time, composable with other programs.
-`IO` values compose.
+Each step of the comprehension is a small program, and the resulting `program` is a composition of all those small steps,
+which you can still compose with other programs. `IO` values compose.
```
Comment (ChristopherDavenport) [inline site/src/main/tut/datatypes/io.md]: `which is compositional with other programs.`?

File: site/src/main/tut/datatypes/io.md (hunk 1 above)
Comment (jcranky) [reply site/src/main/tut/datatypes/io.md]: I'm not sure I get the point.

Comment (alexandru) [reply site/src/main/tut/datatypes/io.md]: It's just English. Whatever is correct and feels good in the moment. No reason to dwell on it too much.

Comment (durban) [reply site/src/main/tut/datatypes/io.md]: `comprehention` -> `comprehension`

Comment (jcranky) [reply site/src/main/tut/datatypes/io.md]: Well I just added a period there because the _for example_ felt a bit awkward before :)

Comment (jcranky) [reply site/src/main/tut/datatypes/io.md]: @durban where?

Comment (jcranky) [reply site/src/main/tut/datatypes/io.md]: Ah, found it. will fix.

**Pull request merged**